### DISCUSSION TITLE
Merge policy

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#3337] [General]
Discussion to track the merge policy for crates in the org because I forget at times.

Bus factor is out of scope, this is just a transparency thing and to remind me what I'm supposed to do and not do.

We have different merge policies per repository. For example `rust-bitcoin` currently has the strictest merge policy, 2-acks with a bunch of carve outs.

Also, which repos should use the merge script (requires usage of ACK xxxxx).

| repository | required # of acks  | additional rules | uses merge script | Who can merge |
|-----------------|----------------------------|-------------------------|----------------------------|------------------------|
| bitcoin        | 2 ** | Tobin merges bot PRs | yes | Any maintainer (mostly Andrew) |
| bitcoinconsensus | 1 | - | yes | Any maintainer (mostly Tobin) |
| bech32        | 1 | Explicit ack from Clark | ? + | Any maintainer (mostly Andrew) |
| corepc        | 0 | - | no | Any maintainer (mostly Tobin) |
| hex              | ? | - | ? | Any maintainer (mostly Andrew) |
| miniscript  | 2 | - | yes | Andrew and Sanket |
| secp             | 1 | - | yes | Any maintainer (mostly Andrew) |

** `rust-bitcoin` has abunch of carve-outs to the 2-ack policy

**Comment by tcharding** [comment]
It was wondering if I should merge https://github.com/rust-bitcoin/hex-conservative/pull/102 that inspired this post.

**Reply by Kixunil** [reply]
It's a simple crate, there's not much that can go wrong, so I wouldn't stress about it requiring many acks. But it'd be nice to setup bors to make sure the history is clean.

**Comment by tcharding** [comment]
Also incase its not known or obvious, Andrew is the original author and BDFL of `rust-bitcoin`. Kix is the original author of `hex`. Clark is the original author of `bech32`. Its a team effort , but we try to remember the original authors of crates/repos (even if the crates move repos).

(I can't speak for `miniscript` or `secp256k1`.)

**Comment by tcharding** [comment]
Amusing proof we need this discussion: https://github.com/rust-bitcoin/rust-bitcoinconsensus/pull/96#issuecomment-2274335752

**Comment by apoelstra** [comment]
FWIW I think all the "only Andrew"s above can be "any maintainer, but mostly Andrew".

There is some benefit to letting me merge stuff in that I run my local CI script on all the merge commits before pushing them. But it's fairly rare for us to have broken merges and we're quick about fixing them.

**Reply by tcharding** [reply]
Perhaps we can add another column 'Has local CI before merge', which repos do you do, `secp`, `bitocin`, and `miniscript`?

**Reply by apoelstra** [reply]
For Blockstream -- `rust-simplicity`, `secp2561-zkp`, `Simplicity`, `simfony`, `elements`, `elements-miniscript`, `rust-elements`, `rust-secp256k1-zkp`.

For rust-bitcoin -- `hex-conservative`, `ordered`, `bech32`, `bitcoin`, `bitcoinconsensus`, `bitcoincore-rpc` (ok this might not work right now), `bitcoind-json-rpc`, `jsonrpc` (ok archived), `miniscript`, `secp256k1`.

This is my current list.

**Reply by apoelstra** [reply]
That includes every single row in your current table.

**Reply by tcharding** [reply]
Nice! So anyone other than you that is merging stuff should keep this in mind and probably not merge anything that would definitely benefit from your local CI? (Well everything benefits but you know what I mean.)

**Reply by tcharding** [reply]
I'm specifically wondering about `bicoind-json-rpc-`, I had in mind already to get you to merge things that touched `jsonrpc`.

**Reply by apoelstra** [reply]
I wouldn't worry about `bitcoind-json-rpc` for now. When it's more stable and complete we can start insisting I test everything.

**Reply by tcharding** [reply]
Done. Cheers.
